cmd/relay/main.go

```diff
@@ -0,0 +1,494 @@
+				attribute.String("environment", env), // Others
+				attribute.Int64("ID", 1),
+			)),
+		)
+		otel.SetTracerProvider(tp)
+	}
+
+	return nil
+}
+
+func runBigsky(cctx *cli.Context) error {
+	// Trap SIGINT to trigger a shutdown.
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+
+	logger, logWriter, err := cliutil.SetupSlog(cliutil.LogOptions{})
+	if err != nil {
+		return err
+	}
+
+	var inductionTraceLog *slog.Logger
+
+	if cctx.IsSet("trace-induction") {
+		traceFname := cctx.String("trace-induction")
+		traceFout, err := os.OpenFile(traceFname, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+		if err != nil {
+			return fmt.Errorf("%s: could not open trace file: %w", traceFname, err)
+		}
+		defer traceFout.Close()
+		if traceFname != "" {
+			inductionTraceLog = slog.New(slog.NewJSONHandler(traceFout, &slog.HandlerOptions{Level: slog.LevelDebug}))
+		}
+	} else {
+		inductionTraceLog = slog.New(slog.NewTextHandler(io.Discard, &slog.HandlerOptions{Level: slog.Level(999)}))
+	}
+
+	// start observability/tracing (OTEL and jaeger)
+	if err := setupOTEL(cctx); err != nil {
+		return err
+	}
+
+	dburl := cctx.String("db-url")
+	logger.Info("setting up main database", "url", dburl)
+	db, err := cliutil.SetupDatabase(dburl, cctx.Int("max-metadb-connections"))
+	if err != nil {
+		return err
+	}
+	if cctx.Bool("db-tracing") {
+		if err := db.Use(tracing.NewPlugin()); err != nil {
+			return err
+		}
+	}
+	if err := db.AutoMigrate(RelaySetting{}); err != nil {
+		panic(err)
+	}
+
+	// TODO: add shared external cache
+	baseDir := identity.BaseDirectory{
+		SkipHandleVerification: true,
+		SkipDNSDomainSuffixes:  []string{".bsky.social"},
+		TryAuthoritativeDNS:    true,
+	}
+	cacheDir := identity.NewCacheDirectory(&baseDir, cctx.Int("did-cache-size"), time.Hour*24, time.Minute*2, time.Minute*5)
+
+	repoman := repomgr.NewRepoManager(&cacheDir, inductionTraceLog)
+
+	var persister events.EventPersistence
+
+	dpd := cctx.String("disk-persister-dir")
+	if dpd == "" {
+		logger.Info("empty disk-persister-dir, use current working directory")
+		cwd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		dpd = filepath.Join(cwd, "relay-persist")
+	}
+	logger.Info("setting up disk persister", "dir", dpd)
+
+	pOpts := events.DefaultDiskPersistOptions()
+	pOpts.Retention = cctx.Duration("event-playback-ttl")
+	pOpts.TimeSequence = cctx.Bool("time-seq")
+
+	// ensure that time-ish sequence stays consistent within a server context
+	storedTimeSeq, hadStoredTimeSeq, err := getRelaySettingBool(db, "time-seq")
+	if err != nil {
+		return err
+	}
+	if !hadStoredTimeSeq {
+		if err := setRelaySettingBool(db, "time-seq", pOpts.TimeSequence); err != nil {
+			return err
+		}
+	} else {
+		if pOpts.TimeSequence != storedTimeSeq {
+			return fmt.Errorf("time-seq stored as %v but param/env set as %v", storedTimeSeq, pOpts.TimeSequence)
+		}
+	}
+
+	dp, err := events.NewDiskPersistence(dpd, "", db, pOpts)
```

Comment:
huh, I would have assumed we'd be switching to pebble persister instead of disk persister? this will be difficult to change once deployed (I assume) so important to decide soon

Reply:
the big thing pebble persister really did was be _new_ code without any cruft, and it doesn't re-sequence events which is what rainbow wants. I think DiskPersistence is still efficient and good.